compiler/hash-types/src/new/fns.rs

```diff
@@ -125,18 +128,93 @@ pub struct FnCallTerm {
     ///
     /// Implicit function calls look like `A<B>`, where as explicit function
     /// calls look like `A(B)`.
-    pub is_implicit: bool,
+    pub implicit: bool,
     // @@Design: optional conditions
 }
 
-impl Display for WithEnv<'_, FnDefId> {
-    fn fmt(&self, _f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        todo!()
+impl Display for WithEnv<'_, &FnTy> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.value.is_unsafe {
+            write!(f, "unsafe ")?;
+        }
+        if self.value.pure && !self.value.implicit {
+            write!(f, "pure ")?;
+        }
+        if !self.value.pure && self.value.implicit {
+            write!(f, "impure ")?;
+        }
+
```

Comment:
`<A> -> B` is pure, `pure (A) -> B` is pure, `impure <A> -> B` is impure, `(A) -> B` is impure

Reply:
Surely, then it should be standardised to:
```
pure <A> -> B   (pure)
<A> -> B        (impure)
pure (A) -> B   (pure)
(A) -> B        (impure)
```
?

Or alternatively, we should document the specifics of this notation to avoid this kind of confusion.